addons/account/models/account_move.py

```diff
@@ -1683,17 +1714,43 @@ def _inverse_partner_id(self):
 
     @api.onchange('company_id')
     def _inverse_company_id(self):
+        self.exchange_rate = False
         self._conditional_add_to_compute('journal_id', lambda m: (
             not m.journal_id.filtered_domain(self.env['account.journal']._check_company_domain(m.company_id))
         ))
 
     @api.onchange('currency_id')
     def _inverse_currency_id(self):
+        self.exchange_rate = False
         (self.line_ids | self.invoice_line_ids)._conditional_add_to_compute('currency_id', lambda l: (
             l.move_id.is_invoice(True)
             and l.move_id.currency_id != l.currency_id
         ))
 
+    @api.onchange('display_rate')
+    def _inverse_display_rate(self):
+        res = {}
+        for move in self:
+            if move.display_rate and move.display_rate != move.system_exchange_rate:
+                move.exchange_rate = move.display_rate
+                if move._origin.display_rate and move.display_rate != move._origin.display_rate:
+                    diff = (move._origin.display_rate - move.display_rate) / move._origin.display_rate
```

Comment:
I remember that I already asked about that, but I don't remember your answer:

Why are we comparing the rates with the previous rates and not with what would be expected?

Let's say:
* system rate is 1, no custom rate set yet
* change the custom rate to 10.1: you get a warning. I expect that.
* change the custom rate to 1.01: you get a warning. Why?

It all depends on when you save or not (`_origin`), which bother me.